subsys/mgmt/mcumgr/lib/cmd/stat_mgmt/src/stat_mgmt.c

```diff
@@ -70,15 +70,16 @@ stat_mgmt_show(struct mgmt_ctxt *ctxt)
 	err |= cbor_encode_text_stringz(&ctxt->encoder, "fields");
 	err |= cbor_encoder_create_map(&ctxt->encoder, &map_enc, CborIndefiniteLength);
 
-	rc = stat_mgmt_impl_foreach_entry(stat_name, stat_mgmt_cb_encode,
-									  &map_enc);
+	if (err == 0) {
+		rc = stat_mgmt_impl_foreach_entry(stat_name, stat_mgmt_cb_encode, &map_enc);
+	}
 
 	err |= cbor_encoder_close_container(&ctxt->encoder, &map_enc);
```

Comment:
If err is not 0, does the container need to be closed?

Reply:
No. When you return non-MGMT_ERR_EOK code from any of handlers, whatever have been placed in response buffer gets overwritten with new "rc" container.